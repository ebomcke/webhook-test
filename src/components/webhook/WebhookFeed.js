import React from 'react';
import { Message, Icon, Feed } from 'semantic-ui-react';
import WebhookItem from './WebhookItem';
import WebhookTester from './WebhookTester';
import { compose } from 'recompose';
import { withWebhooks } from '../../contexts/Webhooks';
import SearchDropdown from './SearchDropdown';

const WebhookFeed = ({ webhooks, filters, updateFilters, endpoint }) => (
  <div>
    {webhooks.length > 0 || filters.length > 0 ? (
      <div>
        <SearchDropdown filters={filters} updateFilters={updateFilters} />
        <Feed>
          {webhooks.map(webhook => (
            <WebhookItem key={webhook.id} webhook={webhook} />
          ))}
        </Feed>
        {webhooks.length == 0 && (
          <Message icon>
            <Icon name="info" />
            <Message.Content>
              <Message.Header>
                I can't find any webhooks with these search criteria
              </Message.Header>
              Try to use less restrictive search criteria.
            </Message.Content>
          </Message>
        )}
      </div>
    ) : (
      <div>
        <Message icon>
          <Icon name="help" />
          <Message.Content>
            <Message.Header>
              Looks like I haven't received any webhook yet
            </Message.Header>
            If you'd like to test that everything works fine you can send a test
            webhook using the form below.
          </Message.Content>
        </Message>
        <WebhookTester endpoint={endpoint} />
      </div>
    )}
  </div>
);

export default compose(withWebhooks)(WebhookFeed);
