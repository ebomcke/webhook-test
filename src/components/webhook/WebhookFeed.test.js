import React from 'react';
import { create } from 'react-test-renderer';
import WebhookFeed from './WebhookFeed';
jest.mock('../base/Timestamp');

describe('Rendering tests', () => {
  test('Initial rendering - populated list', () => {
    const webhookFeed = create(<WebhookFeed filters={[]} />);
    expect(webhookFeed.toJSON()).toMatchSnapshot();
  });

  test('Initial rendering - empty list', () => {
    const webhookFeed = create(<WebhookFeed webhooks={[]} filters={[]} />);
    expect(webhookFeed.toJSON()).toMatchSnapshot();
  });
});
