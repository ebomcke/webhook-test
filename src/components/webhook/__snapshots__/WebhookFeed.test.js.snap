// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests Initial rendering - empty list 1`] = `
<div>
  <div>
    <div
      className="ui icon message"
    >
      <i
        aria-hidden="true"
        className="help icon"
      />
      <div
        className="content"
      >
        <div
          className="header"
        >
          Looks like I haven't received any webhook yet
        </div>
        If you'd like to test that everything works fine you can send a test webhook using the form below.
      </div>
    </div>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="required field"
      >
        <label>
          Url
        </label>
        <div
          aria-expanded={false}
          className="ui search selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            autoComplete="off"
            className="search"
            onChange={[Function]}
            style={
              Object {
                "width": undefined,
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            className="text"
            role="alert"
          />
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
            role="listbox"
          >
            <div
              aria-checked={false}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                /unit-test-org/unit-test-1
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                /unit-test-org/unit-test-2
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <label>
          Body
        </label>
        <div>
          This is a placeholder for the JSON editor.
          <pre>
            <code>
              {"height":"300px","value":"{\\n  \\"status\\": \\"OK\\"\\n}"}
            </code>
          </pre>
        </div>
      </div>
      <button
        className="ui disabled button"
        disabled={true}
        onClick={[Function]}
        tabIndex={-1}
      >
        Send
      </button>
    </form>
  </div>
</div>
`;

exports[`Rendering tests Initial rendering - populated list 1`] = `
<div>
  <div>
    <div
      aria-expanded={false}
      className="ui fluid multiple search selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="combobox"
    >
      <input
        aria-autocomplete="list"
        autoComplete="off"
        className="search"
        onChange={[Function]}
        style={
          Object {
            "width": undefined,
          }
        }
        tabIndex={0}
        type="text"
        value=""
      />
      <span
        className="sizer"
      />
      <div
        aria-live="polite"
        className="default text"
        role="alert"
      >
        Filter by content and/or url
      </div>
      <i
        aria-hidden="true"
        className="search icon"
        onClick={[Function]}
      />
      <div
        aria-multiselectable={true}
        className="menu transition"
        role="listbox"
      >
        <div
          aria-checked={false}
          aria-selected={true}
          className="selected item"
          filtervalue="unit-test-1"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
          type="path"
        >
          <span
            className="text"
          >
            url: /unit-test-1
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          filtervalue="unit-test-2"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
          type="path"
        >
          <span
            className="text"
          >
            url: /unit-test-2
          </span>
        </div>
      </div>
    </div>
    <div
      className="ui feed"
    >
      <div
        className="event"
      >
        <div
          className="label"
        >
          <i
            aria-hidden="true"
            className="code icon"
          />
        </div>
        <div
          className="content"
        >
          <div
            className="summary"
          >
            /unit-test-org/webhook-path-1
            <div
              className="date"
            >
              <div>
                This is a placeholder for the Timestamp.
                <pre>
                  <code>
                    {"timestamp":"webhook-date-1","prefix":"Received"}
                  </code>
                </pre>
              </div>
            </div>
          </div>
          <div
            className="text extra"
          >
            <div
              className="ui inverted segment"
            >
              <pre>
                <code>
                  {
  "unitTest": true
}
                </code>
              </pre>
            </div>
          </div>
          <div
            className="meta"
          >
            <a
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="remove icon"
              />
              Remove
            </a>
          </div>
        </div>
      </div>
      <div
        className="event"
      >
        <div
          className="label"
        >
          <i
            aria-hidden="true"
            className="code icon"
          />
        </div>
        <div
          className="content"
        >
          <div
            className="summary"
          >
            /unit-test-org/webhook-path-2
            <div
              className="date"
            >
              <div>
                This is a placeholder for the Timestamp.
                <pre>
                  <code>
                    {"timestamp":"webhook-date-2","prefix":"Received"}
                  </code>
                </pre>
              </div>
            </div>
          </div>
          <div
            className="text extra"
          >
            <div
              className="ui inverted segment"
            >
              <pre>
                <code>
                  {
  "unitTest": true
}
                </code>
              </pre>
            </div>
          </div>
          <div
            className="meta"
          >
            <a
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="remove icon"
              />
              Remove
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
